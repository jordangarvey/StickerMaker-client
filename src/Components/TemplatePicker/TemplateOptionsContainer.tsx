import React, { FC } from "react";
import { StyleSheet, View } from "react-native";

import Categories from "./Categories";
import Shapes from "./Shapes";

import ColourPicker from "./OptionViews/ColourPicker";
import ExportsButton from "./ExportsButton";
import ShapePicker from "./OptionViews/ShapePicker";
import TextPicker from "./OptionViews/TextPicker";


/**
 * Interface for the Template Options Container props
 */
interface ITemplateOptionsContainerProps {
	currentCategory?: Categories;
	currentOption?: Shapes;
}

/**
 * Component to render the Template Options Container
 */
const TemplateOptionsContainer: FC<ITemplateOptionsContainerProps> = (props) => {
	let options: JSX.Element | null = null;

	switch(props.currentCategory) {
		case Categories.Colour:
				options = <ColourPicker currentOption={props.currentOption as any}/>;
			break;
		case Categories.Shape:
			options = <ShapePicker currentOption={props.currentOption as Shapes}/>;
			break;
		case Categories.Text:
				options = <TextPicker currentOption={props.currentOption as any}/>;
			break;
	}

	return (
		<View style={styles.templateOptionsContainer}>
			{options}

			<ExportsButton/>
		</View>
	);
};

/**
 * Styles for the Template Options Container
 */
const styles = StyleSheet.create({
	templateOptionsContainer: {
		backgroundColor: "#FFFFFF",
		borderColor: "#000000",
		borderWidth: 0.5,
		height: 175,
		width: "100%"
	}
});

export default TemplateOptionsContainer;
