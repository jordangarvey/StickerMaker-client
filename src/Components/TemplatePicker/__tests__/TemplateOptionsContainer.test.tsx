import React from "react";
import renderer from "react-test-renderer";

import TemplateOptionsContainer from "../TemplateOptionsContainer";


describe("The Template Options Container componentâ€™s snapshot", () => {
	it("renders as expected", () => {
		const component = renderer.create(
			<TemplateOptionsContainer/>
		).toJSON();


		expect(component).toMatchSnapshot();
	});
});
