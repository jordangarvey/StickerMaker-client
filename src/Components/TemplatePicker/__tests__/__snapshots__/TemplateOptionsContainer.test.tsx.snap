// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Template Options Container componentâ€™s snapshot renders as expected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#000000",
      "borderWidth": 0.5,
      "height": 175,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#18A0FB",
        "borderBottomLeftRadius": 30,
        "borderBottomRightRadius": 30,
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
        "flexDirection": "row",
        "height": 70,
        "justifyContent": "center",
        "marginHorizontal": 20,
        "opacity": 1,
        "padding": 5,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Export to camera roll
    </Text>
  </View>
</View>
`;
